runtime/android/core_internal/src/org/xwalk/core/internal/HitTestResultInternal.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2006 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// Copyright (c) 2016 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package org.xwalk.core.internal;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.util.Map;
+
+@XWalkAPI(createInternally = true)
+public class HitTestResultInternal {
+    @XWalkAPI
+    public enum type {
+        UNKNOWN_TYPE,
+        ANCHOR_TYPE,
+        PHONE_TYPE,
+        GEO_TYPE,
+        EMAIL_TYPE,
+        IMAGE_TYPE,
+        IMAGE_ANCHOR_TYPE,
+        SRC_ANCHOR_TYPE,
+        SRC_IMAGE_ANCHOR_TYPE,
+        EDIT_TEXT_TYPE
+    }
+
+    private int mType;
+    private String mExtra;
+
+    /**
+     * @hide Only for use by WebViewProvider implementations
+     */
+    public HitTestResultInternal() {
+        mType = 0;
+    }
+
+    /**
+     * @hide Only for use by WebViewProvider implementations
+     */
+    public void setType(int type) {
+        mType = type;
+    }
+
+    /**
+     * @hide Only for use by XWalkViewProvider implementations
+     */
+    public void setExtra(String extra) {
+        mExtra = extra;
+    }
+
+    /**
+     * Gets the type of the hit test result. See the XXX_TYPE constants
+     * defined in this class.
+     *
+     * @return the type of the hit test result
+     * @since 7.0
+     */
+    @XWalkAPI
+    public type getType() {
+        type a;
+        switch (mType){
```

Comment:
@wuhengzhi no, return a, mType is an integer type which is used by internal. we should convert integer to type here.